### PR TITLE
DragDropFiles Control and integration with ListView and FilePicker v3

### PR DESCRIPTION
**Pull request by aaclage**
| Q               | A
| --------------- | ---
| Bug fix?        | [x]
| New feature?    | [x]
| New sample?      | [ ]
| Related issues?  | mentioned in #856

#### What's in this Pull Request?

Update to V3 version.
This new control allows to drag and drop files in pre defined areas.
This control has **High impact** in following controls **ListView** and **FilePicker** since they were changed/added to adapt this new control.
If additional tests or revision need to be made please advise.

This control has the following features that provide enhance and adaptability to existing controls.
- Include drag and drop of file in pre defined area of existing controls.
- Add custom Label on Upload Area.
- Add custom icon based on Office UI Fabric Icons.
- Enable or disable Control.
- Manage the output Files from custom Method "**onDrop**"
- dropEffect manages the action after drop the file. 
      -  https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/dropEffect
- Integration of DragDropFiles in existing Controls **ListView** and **FilePicker** > new Tab "**Upload Files**".
- Control allow drag and drop of Folders with files, only File[] are return with new property "**fullPath**" in file.
      - New property "**fullPath**" was included in file object to allow identify dropped files based on Folders, this allow users to create associated folder path by code (**defensive code included** https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/webkitGetAsEntry)

Sample code:
```
<DragDropFiles 
          dropEffect="copy" 
          enable={true}  
          onDrop={this._getDropFiles}
          iconName="Upload"
          labelMessage= "My custom upload File"
          >
  {/* Specify the components to load where Drag and drop area should work */}
</DragDropFiles>
```

**Simple Sample view:**
image

**Embedded and integrated with ListView:**
image

**Embedded and integrated with FilePicker:**
image

### CODE REVIEW

**Comment by aaclage** [comment]
Hi @AJIXuMuK receive message the branch was changed, 
is this solution under revision? or should not be consider for this version?

Thank you for your attention.

**Comment by AJIXuMuK** [comment]
Hi @aaclage,

The branch has been changed as I merged a PR into the branch.

So your PR is still valid and valuable for `v3` version as well.

I believe @joelfmrodrigues reviewed your PR for `v2` so I will pass the review of this one to him as well :)

**Comment by aaclage** [comment]
Thank you @AJIXuMuK for the feedback. 
I was thinking did something incorrect.

**Comment by joelfmrodrigues** [comment]
@aaclage just some minor things:
I just noticed that you named one of the controls MultiUploadFilePicke**t** and not MultiUploadFilePicke**r** (notice the last character). Any chance you could fix the name? (I'm assuming it was a typo)

File: src/controls/filePicker/MultipleUploadFilePicketTab/MultipleUploadFilePicketTab.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    super(props);
+    this.state = {
+      filePickerResult: undefined
+    };
+  }
+
+  private displayFileNames = (filesResult) => {
+    const result = [];
+    for (var i = 0; i < filesResult.length; i++) {
+      result.push(<div className={styles.localTabFilename}>{filesResult[i].name}</div>);
```
Comment (joelfmrodrigues) [inline src/controls/filePicker/MultipleUploadFilePicketTab/MultipleUploadFilePicketTab.tsx:27]: This loop is generating a warning in the browser console as the repeated elements don't have a key property
```suggestion
      result.push(<div key={i.toString()} className={styles.localTabFilename}>{filesResult[i].name}</div>);
```

Comment (joelfmrodrigues) [reply src/controls/filePicker/MultipleUploadFilePicketTab/MultipleUploadFilePicketTab.tsx:27]: @aaclage I have just added the `key` property to the version in dev-v2 branch

Comment (aaclage) [reply src/controls/filePicker/MultipleUploadFilePicketTab/MultipleUploadFilePicketTab.tsx:27]: @joelfmrodrigues already making the changes.

**Comment by aaclage** [comment]
> @aaclage just some minor things:
> I just noticed that you named one of the controls MultiUploadFilePicke**t** and not MultiUploadFilePicke**r** (notice the last character). Any chance you could fix the name? (I'm assuming it was a typo)

yes, making the change.

**Comment by aaclage** [comment]
@joelfmrodrigues updated "Rename to MultipleUploadFilePickerTab" version for revision.

**Comment by joelfmrodrigues** [comment]
@aaclage this is now merged. Many thanks!

**Comment by aaclage** [comment]
Thank you @joelfmrodrigues :), one small missing detail is to include PR in "Milestone" 3.1.0 .

**Comment by AJIXuMuK** [comment]
@aaclage it won't be missed without milestone anyway :)
Thank you for the contribution!

**Pull request merged**